scio-core/src/main/scala/com/spotify/scio/ScioResult.scala

```diff
@@ -54,7 +59,42 @@ class ScioResult private[scio] (val internal: PipelineResult,
   def accumulatorValuesAtSteps[T](acc: Accumulator[T]): Map[String, T] =
     getAggregatorValues(acc).flatMap(_.getValuesAtSteps.asScala).toMap
 
+  /** Materialize accumulators of finished pipeline to specified file. */
+  def saveMetrics(filename: String): Unit = {
+    require(isCompleted, "Pipeline has to be finished to dump accumulators.")
+
+    val mapper = new ObjectMapper()
+    mapper.registerModule(DefaultScalaModule)
+
+    val out = IOChannelUtils.create(filename, MimeTypes.TEXT)
+
+    try {
+      val totalValues = accumulators
+        .map(acc => (acc, accumulatorTotalValue(acc)))
+        .map(e => AccumulatorValue(e._1.name, e._2))
+
+      val stepsValues = accumulators
+        .map(acc => (acc, accumulatorValuesAtSteps(acc)))
+        .map(e =>
+          AccumulatorStepsValue(e._1.name, e._2.map(a => AccumulatorStepValue(a._1, a._2))))
+
+      val metrics = Metrics(1, AccumulatorMetrics(totalValues, stepsValues))
+      out.write(ByteBuffer.wrap(mapper.writeValueAsBytes(metrics)))
+    } finally {
+      if (out != null) {
+        out.close()
+      }
+    }
+  }
+
   private def getAggregatorValues[T](acc: Accumulator[T]): Iterable[AggregatorValues[T]] =
     aggregators(acc.name).map(a => internal.getAggregatorValues(a.asInstanceOf[Aggregator[_, T]]))
 
 }
+
+private[scio] case class Metrics(version: Int, accumulators: AccumulatorMetrics)
```

Comment:
Do we wanna persist some job info as well? Like job name, ID, scio and scala version? And in that case can `scioVersion` replace `version: Int` here?
It'll make batch processing easier if we save all metrics to the same path.